src/video/SDL_video.c

```diff
@@ -5297,7 +5297,7 @@ bool SDL_StartTextInputWithProperties(SDL_Window *window, SDL_PropertiesID props
     }
 
     // Show the on-screen keyboard, if desired
-    if (AutoShowingScreenKeyboard() && !SDL_ScreenKeyboardShown(window)) {
+    if (AutoShowingScreenKeyboard()) {
```

Comment:
Why was the `SDL_ScreenKeyboardShown()` check originally here? It was correct for `SDL_StartTextInput()`, but now that `SDL_StartTextInputWithProperties()` is a thing, this check no longer makes sense.

If the `props` have changed, `ShowScreenKeyboard` (or a new update method) should be called to communicate the updated keyboard type to the platform layer.